File: typescript/raindrops/raindrops.ts
export function convert(drops: number): string {
  let statement = ''
  if(drops % 3 === 0) statement += 'Pling'
  if(drops % 5 === 0) statement += 'Plang'
  if(drops % 7 === 0) statement += 'Plong'
  return statement.length > 0 ? statement : String(drops)
}

File: ruby/raindrops/raindrops.rb
module Raindrops
  def self.convert(drops)
    sounds = { "Pling" => 3, "Plang" => 5, "Plong" => 7 }
    statement = sounds.reduce("") do |result, (sound, factor)|
      result += sound if (drops % factor).zero?
      result
    end
    statement.empty? ? drops.to_s : statement
  end
end
